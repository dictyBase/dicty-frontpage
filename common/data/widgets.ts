const widgets = [
  {
    link: "/dictyaccess",
    name: "dictyAccess",
  },
  {
    link:
      "http://dictybase.org/tools/jbrowse/?data=data/jbrowse/discoideum&loc=6:1..50011&tracks=reference,gene,transcript",
    name: "Genome Browser",
  },
  {
    link: "/gene/DDB_G0283267",
    name: "Gene Page (in progress)",
  },
  {
    link: "/stockcenter",
    name: "Dicty Stock Center",
  },
]

export default widgets
