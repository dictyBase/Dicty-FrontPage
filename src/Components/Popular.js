// @flow
import React from "react"
import { Flex } from "grid-styled"

import {
  PopularHeader,
  WidgetListBox,
  ListItems,
  WidgetLink,
  WidgetImg,
} from "./styles"

const Popular = (props: {
  /** List of widget items */
  widgets: Array<{
    "link": string,
    "image": any,
    "alt": string,
  }>,
}) => {
  const widgetlist = props.widgets.map((widget, index) =>
    <ListItems key={index}>
        <WidgetLink
              href={widget.link}
              alt={widget.alt}>
              <WidgetImg src={widget.image} />
        </WidgetLink>
    </ListItems>,
  )

  return (
    <Flex wrap>
      <PopularHeader>
            Most popular tools and sections
      </PopularHeader>
      <WidgetListBox>
        {widgetlist}
      </WidgetListBox>
    </Flex>
  )
}

export default Popular
