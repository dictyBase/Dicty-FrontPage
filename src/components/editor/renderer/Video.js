import React, { PureComponent } from "react"

export default class Video extends PureComponent {
  onClick = e => {
    e.stopPropagation()
  }

  render() {
    const { node, isSelected, editor, attributes } = this.props
    const src = node.data.get("src")
    const url = node.data.get("url")

    const { readOnly } = editor.props

    const wrapperStyle = {
      position: "relative",
      paddingBottom: "50.66%",
      paddingTop: "25px",
      height: "0",
      outline: isSelected ? `2px solid #0017ff` : "none",
    }

    const maskStyle = {
      display: isSelected ? "none" : "block",
      position: "absolute",
      top: "0px",
      left: "0px",
      right: "0px",
      bottom: "0px",
      height: "100%",
      cursor: "cell",
      zIndex: 1,
    }

    const iframeStyle = {
      position: "absolute",
      top: "0px",
      left: "0px",
      width: "100%",
      height: "100%",
    }

    return (
      <div>
        <div {...attributes} style={wrapperStyle}>
          <div style={readOnly ? {} : maskStyle} />
          <iframe
            title="video-embed"
            id="ytplayer"
            type="text/html"
            width="580"
            height="390"
            src={src}
            frameBorder="0"
            style={iframeStyle}
          />
        </div>
      </div>
    )
  }
}
