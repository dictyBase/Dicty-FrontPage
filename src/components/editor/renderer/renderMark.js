// @flow
import React, { type Node } from "react"
import { type Mark } from "slate"
import MARKS from "components/editor/constants/marks"

type Props = {
  children: Node,
  mark: Mark,
}

const renderMark = (props: Props) => {
  const { children, mark } = props
  switch (mark.type) {
    case MARKS.BOLD:
      return <strong>{children}</strong>

    case MARKS.ITALIC:
      return <em>{children}</em>

    case MARKS.UNDERLINE:
      return <u>{children}</u>

    case MARKS.STRIKETHROUGH:
      return <del>{children}</del>

    default:
      return null
  }
}

export default renderMark
