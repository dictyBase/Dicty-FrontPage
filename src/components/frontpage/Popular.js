// @flow
import React from "react"
import { Flex, Box } from "rebass"

import {
  PopularHeader,
  WidgetListBox,
  ListItems,
  WidgetLink,
  WidgetImg,
} from "styles"

/** Widget that displays the most popular tools and sections */

const Popular = (props: {
  /** List of widget items */
  widgets: Array<{
    link: string,
    image: any,
    alt: string,
  }>,
}) => {
  const widgetlist = props.widgets.map((widget, index) => (
    <ListItems key={index}>
      <WidgetLink to={widget.link} alt={widget.alt}>
        <WidgetImg src={widget.image} alt={widget.alt} />
      </WidgetLink>
    </ListItems>
  ))

  return (
    <Flex justify={"center"}>
      <Box>
        <PopularHeader>Most popular tools and sections</PopularHeader>
        <WidgetListBox>{widgetlist}</WidgetListBox>
      </Box>
    </Flex>
  )
}

export default Popular
