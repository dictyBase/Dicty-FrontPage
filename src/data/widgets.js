import widgetTECHNIQUES from "images/frontWidgets/widgetTECHNIQUES.png"
import widgetGBROWSER from "images/frontWidgets/widgetGBROWSER.png"
import widgetLEARNDICTY from "images/frontWidgets/widgetLEARNDICTY.png"
import widgetJOBS from "images/frontWidgets/widgetJOBS.png"
import widgetDICTYEXPRESS from "images/frontWidgets/widgetDICTYEXPRESS.png"

export default [
  {
    link: "/research/techniques",
    image: widgetTECHNIQUES,
    alt: "techniques",
  },
  {
    link: "/tools/jbrowse",
    image: widgetGBROWSER,
    alt: "genome browser",
  },
  {
    link: "/explore/learn",
    image: widgetLEARNDICTY,
    alt: "learn dicty",
  },
  {
    link: "/community/jobs",
    image: widgetJOBS,
    alt: "jobs @ dicty",
  },
  {
    link: "/dictyexpress",
    image: widgetDICTYEXPRESS,
    alt: "dicty express",
  },
]
