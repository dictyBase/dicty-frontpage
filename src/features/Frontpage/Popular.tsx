import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import Button from "@material-ui/core/Button"

const useStyles = makeStyles({
  button: {
    width: "100%",
    marginBottom: "5px",
    padding: "25px",
    fontSize: "16px",
    color: "#e5e5e5",
    textTransform: "none",
    backgroundColor: "#004080",
    "&:hover": {
      backgroundColor: "#cce6ff",
      color: "#333232",
    },
  },
  altButton: {
    width: "100%",
    marginBottom: "5px",
    padding: "25px",
    fontSize: "16px",
    textTransform: "none",
    backgroundColor: "#80c1ff",
    color: "#333232",
    "&:hover": {
      backgroundColor: "#cce6ff",
    },
  },
  grid: {
    padding: "8px 16px 8px 16px",
  },
  link: {
    textDecoration: "none",
  },
  listItem: {
    listStyle: "none",
  },
  header: {
    color: "#bdbdbd",
    fontSize: "12px",
    verticalAlign: "top",
    textAlign: "left",
    fontStyle: "italic",
    "@media (max-width: 767px)": {
      fontSize: "14px",
      textAlign: "left",
    },
  },
  innerHeader: {
    textAlign: "center",
  },
  listBox: {
    padding: "0px",
    marginBottom: "5px",
  },
})

/** Widget that displays the most popular tools and sections */

const Popular = () => {
  const classes = useStyles()

  const widgetlist = (
    <>
      <li className={classes.listItem}>
        <a className={classes.link} href="/dictyaccess">
          <Button
            className={classes.button}
            size="small"
            variant="contained"
            color="primary">
            dictyAccess
          </Button>
        </a>
      </li>
      <li className={classes.listItem}>
        <a
          className={classes.link}
          href="http://dictybase.org/tools/jbrowse/?data=data/jbrowse/discoideum&loc=6:1..50011&tracks=reference,gene,transcript">
          <Button
            className={classes.altButton}
            size="small"
            variant="contained"
            color="primary">
            Genome Browser
          </Button>
        </a>
      </li>
      <li className={classes.listItem}>
        <a className={classes.link} href="/gene/gflB">
          <Button
            className={classes.button}
            size="small"
            variant="contained"
            color="primary">
            Gene Page (in progress)
          </Button>
        </a>
      </li>
      <li className={classes.listItem}>
        <a className={classes.link} href="/stockcenter">
          <Button
            className={classes.altButton}
            size="small"
            variant="contained"
            color="primary">
            Dicty Stock Center
          </Button>
        </a>
      </li>
    </>
  )

  return (
    <Grid container justify="center" className={classes.grid}>
      <Grid item xs={12}>
        <div className={classes.header}>
          <div className={classes.innerHeader}>New features and pages</div>
        </div>
        <ul className={classes.listBox}>{widgetlist}</ul>
      </Grid>
    </Grid>
  )
}

export default Popular
