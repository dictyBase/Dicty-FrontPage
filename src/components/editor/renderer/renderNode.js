// @flow
import React, { type Node } from "react"
import { type Block } from "slate"
import styled from "styled-components"
import Video from "components/editor/renderer/Video"
import BLOCKS from "components/editor/constants/blocks"
import INLINES from "components/editor/constants/inlines"

type Props = {
  node: Block,
  children: Node,
  attributes: Object,
}

const Image = styled.img`
  display: block;
  max-width: 100%;
  max-height: 20em;
  box-shadow: ${props => (props.selected ? "0 0 0 2px blue;" : "none")};
`

const renderNode = (props: Props) => {
  const { attributes, children, node } = props
  switch (node.type) {
    case BLOCKS.UL_LIST:
      return <ul {...attributes}>{children}</ul>

    case BLOCKS.OL_LIST:
      return <ol {...attributes}>{children}</ol>

    case BLOCKS.LIST_ITEM:
      return <li {...attributes}>{children}</li>

    case BLOCKS.HEADING_1:
      return <h1 {...attributes}>{children}</h1>

    case BLOCKS.HEADING_2:
      return <h2 {...attributes}>{children}</h2>

    case BLOCKS.HEADING_3:
      return <h3 {...attributes}>{children}</h3>

    case BLOCKS.BLOCKQUOTE:
      return <blockquote {...attributes}>{children}</blockquote>

    case INLINES.LINK: {
      const { data } = node
      const href = data.get("href")
      return (
        <a {...attributes} href={href}>
          {children}
        </a>
      )
    }

    case BLOCKS.IMAGE: {
      const { data, isSelected } = node
      const src = data.get("src")
      const className = isSelected ? "selected" : "unselected"
      const style = { display: "inline-block" }
      return (
        <img
          src={src}
          className={className}
          {...attributes}
          style={style}
          alt=""
        />
      )
    }

    case BLOCKS.ALIGN_LEFT:
      return (
        <div style={{ textAlign: "left" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_CENTER:
      return (
        <div style={{ textAlign: "center" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_RIGHT:
      return (
        <div style={{ textAlign: "right" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.HR:
      return <hr {...attributes} />

    case BLOCKS.TABLE:
      return (
        <table {...attributes}>
          <tbody>{children}</tbody>
        </table>
      )

    case BLOCKS.TABLE_ROW:
      return <tr {...attributes}>{children}</tr>

    case BLOCKS.TABLE_CELL:
      return <td {...attributes}>{children}</td>

    case BLOCKS.VIDEO:
      return <Video {...props} />

    default:
      return null
  }
}

export default renderNode
