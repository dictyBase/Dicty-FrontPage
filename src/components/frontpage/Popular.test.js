import React from "react"
import renderer from "react-test-renderer"
import Popular from "components/frontpage/Popular"
import widgets from "data/widgets"
import "jest-styled-components"

test("Popular snapshot test", () => {
  const component = renderer.create(<Popular widgets={widgets} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
