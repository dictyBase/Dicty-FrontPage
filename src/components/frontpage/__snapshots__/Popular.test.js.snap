// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular snapshot test 1`] = `
<div
  className="sc-bdVaJa bDWFJH sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-bwzfXH uTxCW sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="sc-laTMn dBVuoz"
    >
      Most popular tools and sections
    </div>
    <ul
      className="sc-TFwJa hUWmAw"
    >
      <li
        className="sc-bHwgHz jiapwT"
      >
        <a
          alt="techniques"
          className="sc-bMvGRv dnfvUU"
          href="#"
        >
          <img
            className="sc-kUaPvJ HrdCt"
            src="widgetTECHNIQUES.png"
          />
        </a>
      </li>
      <li
        className="sc-bHwgHz jiapwT"
      >
        <a
          alt="genome browser"
          className="sc-bMvGRv dnfvUU"
          href="#"
        >
          <img
            className="sc-kUaPvJ HrdCt"
            src="widgetGBROWSER.png"
          />
        </a>
      </li>
      <li
        className="sc-bHwgHz jiapwT"
      >
        <a
          alt="learn dicty"
          className="sc-bMvGRv dnfvUU"
          href="/explore/learn"
        >
          <img
            className="sc-kUaPvJ HrdCt"
            src="widgetLEARNDICTY.png"
          />
        </a>
      </li>
      <li
        className="sc-bHwgHz jiapwT"
      >
        <a
          alt="dicty stats"
          className="sc-bMvGRv dnfvUU"
          href="#"
        >
          <img
            className="sc-kUaPvJ HrdCt"
            src="widgetSTATS.png"
          />
        </a>
      </li>
      <li
        className="sc-bHwgHz jiapwT"
      >
        <a
          alt="jobs @ dicty"
          className="sc-bMvGRv dnfvUU"
          href="#"
        >
          <img
            className="sc-kUaPvJ HrdCt"
            src="widgetJOBS.png"
          />
        </a>
      </li>
      <li
        className="sc-bHwgHz jiapwT"
      >
        <a
          alt="dicty express"
          className="sc-bMvGRv dnfvUU"
          href="#"
        >
          <img
            className="sc-kUaPvJ HrdCt"
            src="widgetDICTYEXPRESS.png"
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
